pkg/kubelet/kuberuntime/security_context.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kuberuntime
+
+import (
+	"k8s.io/kubernetes/pkg/api"
+	runtimeapi "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
+)
+
+// determineEffectiveSecurityContext gets container's security context from api.Pod and container.SecurityContext.
+func (m *kubeGenericRuntimeManager) determineEffectiveSecurityContext(pod *api.Pod, sc *api.SecurityContext) *runtimeapi.SecurityContext {
+	effectiveSc := m.securityContextFromPodSecurityContext(pod)
+	if effectiveSc == nil && sc == nil {
+		return nil
+	}
+	if effectiveSc != nil && sc == nil {
+		return effectiveSc
+	}
+	if effectiveSc == nil && sc != nil {
+		return convertToRuntimeSecurityContext(sc)
+	}
+
+	if sc.SELinuxOptions != nil {
+		effectiveSc.SelinuxOptions = convertToRuntimeSELinuxOption(sc.SELinuxOptions)
+	}
+
+	if sc.Capabilities != nil {
+		effectiveSc.Capabilities = convertToRuntimeCapabilities(sc.Capabilities)
+	}
+
+	if sc.Privileged != nil {
+		effectiveSc.Privileged = sc.Privileged
+	}
+
+	if sc.RunAsUser != nil {
+		effectiveSc.RunAsUser = sc.RunAsUser
+	}
+
+	if sc.RunAsNonRoot != nil {
+		effectiveSc.RunAsNonRoot = sc.RunAsNonRoot
+	}
+
+	if sc.ReadOnlyRootFilesystem != nil {
+		effectiveSc.ReadonlyRootfs = sc.ReadOnlyRootFilesystem
+	}
+
+	return effectiveSc
+}
+
+// securityContextFromPodSecurityContext gets security context from api.Pod.
+func (m *kubeGenericRuntimeManager) securityContextFromPodSecurityContext(pod *api.Pod) *runtimeapi.SecurityContext {
```

Comment:
You don't need this method - just resolve the pod security context to a container security context, and then use the same conversion.